File: C++/1040.cpp
#include <iostream>
#include <iomanip>

using namespace std;
 
int main() {
 
    float a,b,c,d,m;
    cin >> a >> b >> c >> d;
    a*=2;
    b*=3;
    c*=4;
    m=(a+b+c+d)/10;
    cout << "Media: " << fixed << setprecision(1) << m << endl;
    
    if(m>=7){
        cout << "Aluno aprovado." << endl;
    }else if(m<5){
        cout << "Aluno reprovado." << endl;
    } else if(m>=5 && m<=6.9){
        cout << "Aluno em exame." << endl;
        float n2,m2;
        cin >> n2;
        cout << "Nota do exame: " << fixed << setprecision(1) << n2 << endl;
        m2=(m+n2)/2;
        if(m2>=5){
            cout << "Aluno aprovado." << endl;
        }else if(m2<=4.9){
            cout << "Aluno reprovado." << endl;
        }
        cout << "Media final: " << fixed << setprecision(1) << m2 << endl;
    }
    return 0;
}

File: Java/1040.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
public class Main {
    public static void main(String[] args) throws IOException {
        InputStreamReader ir = new InputStreamReader(System.in);
        BufferedReader in = new BufferedReader(ir);
        String line = in.readLine();
        String[] values = line.trim().split(" ");
        float a=Float.parseFloat(values[0]);
        float b=Float.parseFloat(values[1]);
        float c=Float.parseFloat(values[2]);
        float d=Float.parseFloat(values[3]);
        float notaEmExame;
        double media = (a*2+b*3+c*4+d*1)/10;
        System.out.printf("Media: %.1f\n",media);
        if(media >= 7.0){
            System.out.printf("Aluno aprovado.\n");
        }else if(media < 5.0){
            System.out.printf("Aluno reprovado.\n");
        }else {
            System.out.printf("Aluno em exame.\n");
            notaEmExame = Float.parseFloat(in.readLine());
            System.out.printf("Nota do exame: %.1f\n",notaEmExame);
            media=(media+notaEmExame) / 2;
            if(media >= 5.0){
                System.out.printf("Aluno aprovado.\n");
            }else{
                System.out.printf("Aluno reprovado.\n");
            }
            System.out.printf("Media final: %.1f\n",media);
        }
    }
}